control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -391,39 +407,212 @@ func (r Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkainterna
+	} else if err != nil {
+		logging.FromContext(ctx).Errorw("Unable to get an existing ScaledObject", zap.Error(err))
+		return err
+	} else if !metav1.IsControlledBy(scaledObject, obj) {
+		return fmt.Errorf("ScaledObject %q is not owned by %q", scaledObject.Name, obj)
+	} else if !equality.Semantic.DeepDerivative(scaledObject.Spec, expectedScaledObject.Spec) {
+		logging.FromContext(ctx).Debug(fmt.Sprintf("ScaledObject changed, found: %#v expected: %#v", scaledObject.Spec, expectedScaledObject.Spec))
+		scaledObject.Spec = expectedScaledObject.Spec
+		if _, err = r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Update(ctx, scaledObject, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return kedafunc.ScaleObjectUpdated(scaledObject.Namespace, scaledObject.Name)
+	} else {
+		logging.FromContext(ctx).Debugw("Reusing existing ScaledObject", zap.Any("ScaledObject", scaledObject))
+	}
+
+	return nil
+}
+
+func (r *Reconciler) reconcileTriggerAuthentication(ctx context.Context, expectedTriggerAuth *kedav1alpha1.TriggerAuthentication, obj metav1.Object) error {
+	triggerAuth, err := r.KedaClient.KedaV1alpha1().TriggerAuthentications(expectedTriggerAuth.Namespace).Get(ctx, expectedTriggerAuth.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		triggerAuth, err = r.KedaClient.KedaV1alpha1().TriggerAuthentications(expectedTriggerAuth.Namespace).Create(ctx, expectedTriggerAuth, metav1.CreateOptions{})
+		if err != nil {
+			return reconciler.NewEvent(corev1.EventTypeWarning, "TriggerAuthenticationFailed", "TriggerAuthentication deployment failed to: \"%s/%s\", %w",
+				expectedTriggerAuth.Namespace, expectedTriggerAuth.Name, err)
+		}
+		return reconciler.NewEvent(corev1.EventTypeNormal, "TriggerAuthenticationCreated", "TriggerAuthentication created: \"%s/%s\"",
+			triggerAuth.Namespace, triggerAuth.Name)
+	} else if err != nil {
+		logging.FromContext(ctx).Errorw("Unable to get an existing ScaledObject", zap.Error(err))
+		return err
+	} else if !metav1.IsControlledBy(triggerAuth, obj) {
+		return fmt.Errorf("ScaledObject %q is not owned by %q", triggerAuth.Name, obj)
+	} else if !equality.Semantic.DeepDerivative(triggerAuth.Spec, expectedTriggerAuth.Spec) {
+		logging.FromContext(ctx).Debug(fmt.Sprintf("TriggerAuthentication changed, found: %#v expected: %#v", triggerAuth.Spec, expectedTriggerAuth.Spec))
+		triggerAuth.Spec = expectedTriggerAuth.Spec
+		if _, err = r.KedaClient.KedaV1alpha1().TriggerAuthentications(expectedTriggerAuth.Namespace).Update(ctx, triggerAuth, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return reconciler.NewEvent(corev1.EventTypeNormal, "TriggerAuthenticationUpdated", "TriggerAuthentication updated: \"%s/%s\"",
+			triggerAuth.Namespace, triggerAuth.Name)
+	} else {
+		logging.FromContext(ctx).Debugw("Reusing existing TriggerAuthentication", zap.Any("TriggerAuthentication", triggerAuth))
+	}
+
+	return nil
```

Comment:
Same here, can we stay with the convention of returning early when there is an error and avoid using else ifs with returns?